File: Tugas12/Perpustakaan_Web_API/Pinjam.php
<?php
//Simpanlah dengan nama file : Pinjam.php
require_once 'database.php';
class Pinjam 
{
    private $db;
    private $table = 'pinjam';
    public $nomorbukti = "";
    public $tglpinjam = "";
    public $kodeanggota = "";
    public $kodebuku1 = "";
    public $kodebuku2 = "";
    public $tglhrskembali = "";
    public $tgldikembalikan = "";
    public $statuspinjam = "";
    public $denda = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id"; 
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_nomorbukti(string $nomorbukti) 
    {
        $query = "SELECT * FROM $this->table WHERE  nomorbukti = '$nomorbukti'"; // Menambahkan tanda kutip pada nilai $nomorbukti
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`nomorbukti`,`tglpinjam`,`kodeanggota`,`kodebuku1`,`kodebuku2`,`tglhrskembali`,`tgldikembalikan`,`statuspinjam`,`denda`) VALUES ('$this->nomorbukti','$this->tglpinjam','$this->kodeanggota','$this->kodebuku1','$this->kodebuku2','$this->tglhrskembali','$this->tgldikembalikan','$this->statuspinjam','$this->denda')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET nomorbukti = '$this->nomorbukti', tglpinjam = '$this->tglpinjam', kodeanggota = '$this->kodeanggota', kodebuku1 = '$this->kodebuku1', kodebuku2 = '$this->kodebuku2', tglhrskembali = '$this->tglhrskembali', tgldikembalikan = '$this->tgldikembalikan', statuspinjam = '$this->statuspinjam', denda = '$this->denda' 
        WHERE idpinjam = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_nomorbukti(string $nomorbukti): int
    {
        $query = "UPDATE $this->table SET nomorbukti = '$nomorbukti', tglpinjam = '$this->tglpinjam', kodeanggota = '$this->kodeanggota', kodebuku1 = '$this->kodebuku1', kodebuku2 = '$this->kodebuku2', tglhrskembali = '$this->tglhrskembali', tgldikembalikan = '$this->tgldikembalikan', statuspinjam = '$this->statuspinjam', denda = '$this->denda' 
        WHERE nomorbukti = '$nomorbukti'"; // Ganti <nama_kolom> dan <nilai_kolom> sesuai dengan nama kolom dan nilai yang sesuai. 
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE idpinjam = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_nomorbukti(string $nomorbukti): int
    {
        $query = "DELETE FROM $this->table WHERE nomorbukti = $nomorbukti";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>

File: Tugas12/Perpustakaan_Client_API/Pinjam.py
import requests
import json
class Pinjam:
    def __init__(self):
        self.__id=None
        self.__nomorbukti = None
        self.__tglpinjam = None
        self.__kodeanggota = None
        self.__kodebuku1 = None
        self.__kodebuku2 = None
        self.__tglhrskembali = None
        self.__tgldikembalikan = None
        self.__statuspinjam = None
        self.__denda = None
        self.__url = "http://localhost/perpustakaan/pinjam_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def nomorbukti(self):
        return self.__nomorbukti
        
    @nomorbukti.setter
    def nomorbukti(self, value):
        self.__nomorbukti = value
    @property
    def tglpinjam(self):
        return self.__tglpinjam
        
    @tglpinjam.setter
    def tglpinjam(self, value):
        self.__tglpinjam = value
    @property
    def kodeanggota(self):
        return self.__kodeanggota
        
    @kodeanggota.setter
    def kodeanggota(self, value):
        self.__kodeanggota = value
    @property
    def kodebuku1(self):
        return self.__kodebuku1
        
    @kodebuku1.setter
    def kodebuku1(self, value):
        self.__kodebuku1 = value
    @property
    def kodebuku2(self):
        return self.__kodebuku2
        
    @kodebuku2.setter
    def kodebuku2(self, value):
        self.__kodebuku2 = value
    @property
    def tglhrskembali(self):
        return self.__tglhrskembali
        
    @tglhrskembali.setter
    def tglhrskembali(self, value):
        self.__tglhrskembali = value
    @property
    def tgldikembalikan(self):
        return self.__tgldikembalikan
        
    @tgldikembalikan.setter
    def tgldikembalikan(self, value):
        self.__tgldikembalikan = value
    @property
    def statuspinjam(self):
        return self.__statuspinjam
        
    @statuspinjam.setter
    def statuspinjam(self, value):
        self.__statuspinjam = value
    @property
    def denda(self):
        return self.__denda
        
    @denda.setter
    def denda(self, value):
        self.__denda = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_nomorbukti(self, nomorbukti):
        url = self.__url+"?nomorbukti="+nomorbukti
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['idpinjam']
            self.__nomorbukti = item['nomorbukti']
            self.__tglpinjam = item['tglpinjam']
            self.__kodeanggota = item['kodeanggota']
            self.__kodebuku1 = item['kodebuku1']
            self.__kodebuku2 = item['kodebuku2']
            self.__tglhrskembali = item['tglhrskembali']
            self.__tgldikembalikan = item['tgldikembalikan']
            self.__statuspinjam = item['statuspinjam']
            self.__denda = item['denda']
        return data
    def simpan(self):
        payload = {
            "nomorbukti":self.__nomorbukti,
            "tglpinjam":self.__tglpinjam,
            "kodeanggota":self.__kodeanggota,
            "kodebuku1":self.__kodebuku1,
            "kodebuku2":self.__kodebuku2,
            "tglhrskembali":self.__tglhrskembali,
            "tgldikembalikan":self.__tgldikembalikan,
            "statuspinjam":self.__statuspinjam,
            "denda":self.__denda
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_nomorbukti(self, nomorbukti):
        url = self.__url+"?nomorbukti="+nomorbukti
        payload = {
            "nomorbukti":self.__nomorbukti,
            "tglpinjam":self.__tglpinjam,
            "kodeanggota":self.__kodeanggota,
            "kodebuku1":self.__kodebuku1,
            "kodebuku2":self.__kodebuku2,
            "tglhrskembali":self.__tglhrskembali,
            "tgldikembalikan":self.__tgldikembalikan,
            "statuspinjam":self.__statuspinjam,
            "denda":self.__denda
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_nomorbukti(self, nomorbukti):
        url = self.__url+"?nomorbukti="+nomorbukti
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers) 
        return response.text
